### PR TITLE
Generalize fibonacci rust example - Issue #128

### LINKED ISSUE #128: Update rust fibo example
I wonder why [fibonacci.rs example](https://github.com/privacy-scaling-explorations/chiquito/blob/main/examples/fibonacci.rs) doesn't follow the tutorial style? (fibo_first_step, padding)
I've noticed that there was some work going on #24 but closed. However, I suppose updating the examples would help :)

There is implementation in my branch (https://github.com/jae-cuz/chiquito/commit/3bcafc88fe38fea882fdcd35292d5f0d73994aee), so please have a look!

### PR DESCRIPTION
**Pull request by jae-cuz**
Resolves #128.

@leolara @qwang98 Ready for review. Added comments for added functionalities.

One thing to ask:
If we add a new file `fibo_with_padding.rs`, should we erase the same comments from `fibonacci.rs`? Please advise. Thanks!

### CODE REVIEW

File: examples/fibo_with_padding.rs [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        ctx.pragma_num_steps(11);
+        // constrain steptype of first step
+        ctx.pragma_first_step(&fibo_first_step);
+        // constrain steptype of last step
+        ctx.pragma_last_step(&padding);
+        // Note that because we constrain last step to be padding, the maximum number of
+        // Fibonacci sequence is 10. (one less than num_steps above)
+
+        // Expose the result of calculation and round number
+        ctx.expose(b, chiquito::ast::ExposeOffset::Last);
```
Comment (leolara) [inline examples/fibo_with_padding.rs:167]: @jae-cuz do a `use  chiquito::ast::ExposeOffset`

**Comment by leolara** [comment]
@qwang98 you are the reviewer of this

**Comment by qwang98** [comment]
> @leolara @qwang98 Ready for review. Added comments for added functionalities.
> 
> One thing to ask: If we add a new file `fibo_with_padding.rs`, should we erase the same comments from `fibonacci.rs`? Please advise. Thanks!

I agree that we should remove repetitive comments from `fibonacci.rs` but mention at the top of `fibo_with_padding.rs` that users should go through `fibonacci.rs` first.

**Review by qwang98 — APPROVED** [review]
Please just fix the comments as suggested and I'm good with the PR otherwise.

**Comment by qwang98** [comment]
I'm good with this!

**Pull request merged**